app/src/main/java/com/example/mapsandcollections/dto/CollectionItemModel.java

```diff
@@ -0,0 +1,282 @@
+package com.example.mapsandcollections.dto;
+
+import android.content.Context;
+import android.os.Handler;
+import android.os.Looper;
+
+import com.example.mapsandcollections.R;
+import com.example.mapsandcollections.components.tasker.Tasker;
+
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Random;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+public class CollectionItemModel implements IItemTaskModel {
+
+
+    private String ARRAY_LIST;
+    private String LINKED_LIST;
+    private String COW_LIST;
+
+    private String ADD_START;
+    private String ADD_MIDDLE;
+    private String ADD_END;
+    private String REMOVE_START;
+    private String REMOVE_MIDDLE;
+    private String REMOVE_END;
+    private String SEARCH_BY_VALUE;
+
+    private final Handler handler;
+    private final Context context;
+    private List<Object> arrayList;
+    private List<Object> linkedList;
+    private List<Object> copyOnWriteArrayList;
+    private Tasker.OnTaskDoneListener listener;
+    private List<ItemTask> itemTaskList;
+
+
+    CollectionItemModel(Context context) {
+        this.context = context;
+        handler = new Handler(Looper.getMainLooper());
+        initStrings();
+        createModel();
+    }
+
+    private void initStrings() {
+        ARRAY_LIST = context.getString(R.string.array_list);
+        LINKED_LIST = context.getString(R.string.linked_list);
+        COW_LIST = context.getString(R.string.cow_list);
+        ADD_START = context.getString(R.string.add_start);
+        ADD_MIDDLE = context.getString(R.string.add_middle);
+        ADD_END = context.getString(R.string.add_end);
+        REMOVE_START = context.getString(R.string.remove_start);
+        REMOVE_MIDDLE = context.getString(R.string.remove_middle);
+        REMOVE_END = context.getString(R.string.remove_end);
+        SEARCH_BY_VALUE = context.getString(R.string.serch_by_value);
+
+    }
+
+
+    @Override
+    public List<ItemTask> getItemTasks() {
+        return itemTaskList;
+    }
+
+    @Override
+    public List<Callable<Double>> getTasks(String elements, Tasker.OnTaskDoneListener listener) {
+        setListener(listener);
+        createCollections(elements);
+        final List<Callable<Double>> taskList = new ArrayList<>();
+        for (ItemTask itemTask : getItemTasks()) {
+            taskList.add(itemTask.getTask());
+        }
+        return taskList;
+    }
+
+    private void createModel() {
+
```

Comment:
think how to reduce amount of hardcoded tasks if all collections implements List<T>